### PR TITLE
Update and style Terminal

### PR DESCRIPTION
**Pull request by hatched**
Update and style the terminal to match the new design spec.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

File: eslint-rules/no-single-tests.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
+'use strict';
 module.exports = function(context) {
   return {
     Identifier: function(node) {
-      var nodeName = node.name;
+      // Only run these rules if it's in a test file.
+      if (context.getFilename().indexOf('test') === -1) {
```
Comment (hatched) [inline eslint-rules/no-single-tests.js]: Driveby fix because one of the terminal methods is called `fit` which caused the lint to fail.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8233/

File: jujugui/static/gui/src/app/components/terminal/_terminal.scss [hunk 2]
```diff
@@ -1,17 +1,58 @@
...
+
+    &-actions {
+      position: absolute;
+      top: 0;
+      right: 0;
+      line-height: 36px;
+
+      .svg-icon {
+        stroke: #cdcdcd;
+        fill: #cdcdcd;
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/terminal/_terminal.scss]: This can be `$mid-grey`

File: jujugui/static/gui/src/app/components/terminal/_terminal.scss [hunk 3]
```diff
@@ -1,17 +1,58 @@
...
+        margin-right: 12px;
+        cursor: pointer;
+      }
+
+      .svg-icon:hover {
+        stroke: #f7f7f7;
+        fill: #f7f7f7
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/terminal/_terminal.scss]: This can be `$light-grey`

File: jujugui/static/gui/src/app/components/terminal/terminal.js [hunk 4]
```diff
@@ -83,70 +79,77 @@ class Terminal extends React.Component {
...
+        <div className="juju-shell__header">
+          <span className="juju-shell__header-label">Juju Shell</span>
+          <div className="juju-shell__header-actions">
+            <span onClick={this.setSize.bind(this, 'min')}>
+              <SvgIcon name="minimize-bar_16" size="16" />
+            </span>
+            <span onClick={this.setSize.bind(this, 'max')}>
+              <SvgIcon name="maximize-bar_16" size="16" />
+            </span>
+            <span onClick={this.close.bind(this)}>
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/terminal/terminal.js]: These clickable spans should probably have `tabIndex="0" role="button"` so they can be accessed by the keyboard.

File: jujugui/static/gui/src/app/components/terminal/_terminal.scss (hunk 2 above)
Comment (hatched) [reply jujugui/static/gui/src/app/components/terminal/_terminal.scss]: I can't seem to find this being defined anywhere?

File: jujugui/static/gui/src/app/components/terminal/_terminal.scss (hunk 3 above)
Comment (hatched) [reply jujugui/static/gui/src/app/components/terminal/_terminal.scss]: Yep, I'll update that, thanks.

File: jujugui/static/gui/src/app/components/terminal/terminal.js (hunk 4 above)
Comment (hatched) [reply jujugui/static/gui/src/app/components/terminal/terminal.js]: good call

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8239/

File: jujugui/static/gui/src/app/components/terminal/_terminal.scss (hunk 2 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/terminal/_terminal.scss]: These colours are defined in Vanilla: https://github.com/vanilla-framework/vanilla-framework/blob/v0.1.0/scss/_global-settings.scss

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8245/
